update/v1alpha1/types_update_status_progress.go

```diff
@@ -0,0 +1,424 @@
+	StartedAt metav1.Time `json:"startedAt"`
+
+	// completedAt is the time when the update completed
+	// +optional
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=date-time
+	CompletedAt *metav1.Time `json:"completedAt,omitempty"`
+
+	// estimatedCompletedAt is the estimated time when the update will complete
+	// +optional
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=date-time
+	EstimatedCompletedAt *metav1.Time `json:"estimatedCompletedAt,omitempty"`
+}
+
+// ControlPlaneAssessment is the assessment of the control plane update process
+// +kubebuilder:validation:Enum=Unknown;Progressing;Completed;Degraded
+type ControlPlaneAssessment string
+
+const (
+	// Unknown means the update status and health cannot be determined
+	ControlPlaneAssessmentUnknown ControlPlaneAssessment = "Unknown"
+	// Progressing means the control plane is updating and no problems or slowness are detected
+	ControlPlaneAssessmentProgressing ControlPlaneAssessment = "Progressing"
+	// Completed means the control plane successfully completed updating and no problems are detected
+	ControlPlaneAssessmentCompleted ControlPlaneAssessment = "Completed"
+	// Degraded means the process of updating the control plane suffers from an observed problem
+	ControlPlaneAssessmentDegraded ControlPlaneAssessment = "Degraded"
+)
+
+// ControlPlaneUpdateVersions contains the original and target versions of the upgrade
+type ControlPlaneUpdateVersions struct {
+	// previous is the version of the control plane before the update. When the cluster is being installed
+	// for the first time, the version will have a placeholder value '<none>' and carry 'Installation' metadata
+	// +required
+	// +kubebuilder:validation:XValidation:rule="self.version == '<none>' ? (has(self.metadata) && self.metadata.exists(m, m.key == 'Installation')) : !(has(self.metadata) && self.metadata.exists(m, m.key == 'Installation'))",message="previous version must be '<none>' iff marked with Installation metadata"
+	Previous Version `json:"previous"`
+
+	// target is the version of the control plane after the update. It may never be '<none>' or have `Installation` metadata
+	// +required
+	// +kubebuilder:validation:XValidation:rule="self.version != '<none>' && !(has(self.metadata) && self.metadata.exists(m, m.key == 'Installation'))",message="target version must not be '<none>' or have Installation metadata"
+	Target Version `json:"target"`
+}
+
+// Version describes a version involved in an update, typically on one side of an update edge
+type Version struct {
+	// version is a semantic version string, or a placeholder '<none>' for the special case where this
+	// is a "previous" version in a new installation, in which case the metadata must contain an item
+	// with key 'Installation'
+	// +required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:MinLength=5
+	// +kubebuilder:validation:MaxLength=64
+	Version string `json:"version"`
```

Comment:
Could we validate this using a regex in CEL? Or rather, check that it's `<none>` or matching a valid sem-ver regex

Reply:
I thought I had this :thinking: , right now I'm not sure if I dropped it by mistake or if I had a reason